mesh_nvm.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2017 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include "mbed_trace.h"
+#include "storage-selector.h"
+#include "LittleFileSystem.h"
+#include "ns_file_system.h"
+
+#define TRACE_GROUP "anvm"
+
+#if MBED_CONF_APP_MESH_TYPE == MESH_THREAD
+LittleFileSystem fs("fs");
+BlockDevice* bd = storage_selector();
```

Comment:
How does this work? Is this reliable to be called before `main()` or should you initialize `*bd` only at the `mesh_nvm_initialize()`